weblate/api/views.py

```diff
@@ -726,8 +731,52 @@ def destroy(self, request, *args, **kwargs):
         return super().destroy(request, *args, **kwargs)
 
 
+class CreditsMixin:
+    @action(detail=True, methods=["get"])
+    def credits(self, request, **kwargs):
+        obj = self.get_object()
+
+        try:
+            start_date = from_current_timezone(
+                datetime.fromisoformat(request.query_params["start"])
+            )
+        except (ValueError, MultiValueDictKeyError) as err:
+            raise BadRequest("Invalid format for `start`") from err
+
+        try:
+            end_date = from_current_timezone(
+                datetime.fromisoformat(request.query_params["end"])
+            )
+        except (ValueError, MultiValueDictKeyError) as err:
+            raise BadRequest("Invalid format for `end`") from err
+
+        language = None
+
+        if "lang" in request.query_params:
+            language = request.query_params["lang"]
+
+        kwargs: dict[str, Any]
+        if isinstance(obj, Project):
+            kwargs = {"translation__component__project": obj}
+        elif isinstance(obj, Component):
+            kwargs = {"translation__component": obj}
+        else:
+            raise TypeError("Expected project or component")
+
+        data = generate_credits(
+            None
+            if request.user.has_perm("reports.view", obj) or request.user.is_anonymous
```

Comment:
Anonymous user should not get any reports as it exposes e-mail addresses. That's why regular users get only own report and only users with `reports.view` permission get full one.

```suggestion
            if request.user.has_perm("reports.view", obj)
```

Reply:
Got it. So what should be the behaviour for anonymous users? Should they get a 401 on this route or should the JSON not contain the email fields?

I'm leaning more on the idea of the JSON not having the email fields.

Reply:
I would make it fail with 401.